### ISSUE TITLE
Nested field validation fails

### ISSUE THREAD

**Issue by rahulmg05** [noirbizarre/flask-restplus#355]
Hi,
I have enabled validation on one of my resources using @api.expect(model, validate=True). The model I have has a nested field. The nested field validation fails here.

This is my code snippet.

`from flask_restplus import Namespace, Resource, fields
from flask import request

from core import utils

api = Namespace('todo', description='Resource for all TODO items operations')

todo_model = api.model('todo_model', {
    'name': fields.String(required=True, max_length=1024),
    'description': fields.String(required=True, max_length=8192)
})

todo_details_model = api.model('todo_details', {
    'todos': fields.List(fields.Nested(todo_model), required=True)
})

@api.route('/')
@api.header('X-User', type=str, required=True, description="User name")
class TodoList(Resource):
    def get(self):
        """Gets the list of all todos for a given user"""
        userid = request.headers['X-User']
        return "Success"#utils.get_all_todos_usr(userid)

    @api.expect(todo_details_model, validate=True)
    def post(self):
        """Adds new todos for a given user"""
        userid = request.headers['X-User']
        return "Success" #utils.add_todos_user(userid)`

Though I have set "required=True" for both fields in the todo_model, when trying to validate the following request data the validation returns a success which is not true. 

{
"todos":[]
}

Here the todos key is an empty list.  How do I make sure the validation fails if the list is empty ?

**Comment by pwyssmuller** [comment]
@rahulmg05 have you found a solution to this? I have the exact same problem...

**Comment by rahulmg05** [comment]
@pwyssmuller No, I have not found a solution within the RESTPlus framework. I decided to go ahead and use marshmallow for validation instead. What @api.expect() does is just validate if the key is present. It does not validate the value. So in case of nested fields, the nested data does not get validated.

**Comment by pwyssmuller** [comment]
@rahulmg05 I got around by adding 'min_items=1' as a List parameter. This way it will actually validate the list elements. It my case at least one item is always required, so it worked out.
